server/src/handler/mod.rs

```diff
@@ -16,6 +19,18 @@ impl<Deps: TpmContextDeps> CommandHandler<Deps> {
     pub fn new(crypto: Deps::Crypto) -> Self {
         Self { crypto }
     }
+    /// invariant: buffer will fit into TPM2B_DIGEST which is much less
+    /// than the maximum request of SP 800 90A
+    fn infallable_get_random(&mut self, buffer: &mut [u8]) {
+        if self.crypto.fill_bytes(&[], buffer).is_err() {
+            let mut entropy = <Deps::Crypto as Drbg>::Entropy::default();
+            self.crypto.fill_entropy(&mut entropy);
```

Comment:
Personally I'd rather see two different types, one for entropy source and one for `Drbg`. That would make `Crypto` a plain struct that has both of these types, and make it more clear what you're doing `self.crypto.entropy.fill_entropy(...)`, `self.crypto.drbg.reseed(...)`. I find it to be a tad confusing that that this type is trying to be both of these things at the same time.